strategy/live_trader.py

```
@@ -0,0 +1,714 @@
+        if mstate.max_up_drift >= self.exit_thresh and (mid - 0.50) < self.exit_reversal:
+            mstate.reversal_seen_up = True
+
+        # --- FILL DETECTION (Paper / Live) ---
+        if mstate.status == "IDLE":
+            mstate.status = "QUOTING"
+            mstate.last_action = f"Quoting bids @ {resting_up:.2f} / {resting_down:.2f}"
+
+        if not mstate.pair_captured and not mstate.exit_taken:
+            # UP Leg Fill Check
+            if not mstate.filled_up:
+                if mstate.up_ask is not None and mstate.up_ask <= resting_up:
```

Ruff (0.16.3)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)